lib/db/schema/patch-045-046.sql

```diff
@@ -0,0 +1,224 @@
+        email,
+        uid,
+        emailCode,
+        isVerified,
+        isPrimary,
+        verifiedAt,
+        createdAt
+    )
+    VALUES(
+        LOWER(normalizedEmail),
+        email,
+        uid,
+        emailCode,
+        isVerified,
+        isPrimary,
+        verifiedAt,
+        createdAt
+    );
+
+    COMMIT;
+END;
+
+CREATE PROCEDURE `accountEmails_1` (
+    IN `inUid` BINARY(16)
+)
+BEGIN
+    -- Return all emails that belong to the uid.
+    -- Email from the accounts table is considered the primary email.
+	(SELECT
+        a.normalizedEmail,
+        a.email,
+        a.uid,
+        a.emailCode,
+        a.emailVerified AS isVerified,
+        TRUE AS isPrimary,
+        a.createdAt AS verifiedAt,
+        a.createdAt
+    FROM
+        accounts a
+    WHERE
+        uid = LOWER(inUid))
+
+    UNION ALL
+
+    (SELECT
+        e.normalizedEmail,
+        e.email,
+        e.uid,
+        e.emailCode,
+        e.isVerified,
+        e.isPrimary,
+        e.verifiedAt,
+        e.createdAt
+    FROM
+        emails e
+    WHERE
+        uid = LOWER(inUid))
+    ORDER BY createdAt;
+END;
+
+CREATE PROCEDURE `deleteEmail_1` (
+    IN `inUid` BINARY(16),
+    IN `inNormalizedEmail` VARCHAR(255)
+)
+BEGIN
+    SET @deletedCount = 0;
+    SET @isPrimary = 0;
+
+    -- Don't delete any email that are considered primary emails.
+    -- Since this should be the most common scenario, we try it first
+    DELETE FROM emails WHERE normalizedEmail = inNormalizedEmail AND uid = inUid AND isPrimary = 0;
+
+    -- Check to see if no rows were deleted, if this is the case, check to see if user
+    -- is attempting to deleted the email in the account table.
+    SELECT ROW_COUNT() INTO @deletedCount;
+    IF @deletedCount = 0 THEN
+      SELECT COUNT(*) INTO @isPrimary FROM accounts WHERE normalizedEmail = inNormalizedEmail AND uid = inUid;
+
+      IF @isPrimary = 1 THEN
+        SIGNAL SQLSTATE '45000' SET MESSAGE_TEXT = 'Can not delete a primary email address.';
+      END IF;
+    END IF;
+END;
+
+CREATE PROCEDURE `deleteAccount_12`(
+  IN `uidArg` BINARY(16)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  DELETE FROM sessionTokens WHERE uid = uidArg;
+  DELETE FROM keyFetchTokens WHERE uid = uidArg;
+  DELETE FROM accountResetTokens WHERE uid = uidArg;
+  DELETE FROM passwordChangeTokens WHERE uid = uidArg;
+  DELETE FROM passwordForgotTokens WHERE uid = uidArg;
+  DELETE FROM accounts WHERE uid = uidArg;
+  DELETE FROM devices WHERE uid = uidArg;
+  DELETE FROM unverifiedTokens WHERE uid = uidArg;
+  DELETE FROM unblockCodes WHERE uid = uidArg;
+  DELETE FROM emails WHERE uid = uidArg;
+
+  COMMIT;
+END;
+
+
+CREATE PROCEDURE `verifyEmail_4`(
+    IN `inUid` BINARY(16),
+    IN `inEmailCode` BINARY(16)
+)
+BEGIN
+    SET @updatedCount = 0;
+
+    UPDATE accounts SET emailVerified = true WHERE uid = inUid AND emailCode = inEmailCode;
```

Comment:
If the `emailCode` is not found in either table, should this procedure error out?